pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -79,14 +98,18 @@ func (c *configResource) Process() (reconcile.Result, error) {
 		// For cross site, reconcile must come before compute, because
 		// we need xsite service details to compute xsite struct
 		siteService := &corev1.Service{}
-		if result, err := kube.LookupResource(c.infinispan.GetSiteServiceName(), c.infinispan.Namespace, siteService, c.client, c.log); result != nil {
+		if result, err := kube.LookupResource(c.infinispan.GetSiteServiceName(), c.infinispan.Namespace, siteService, c); result != nil {
 			return *result, err
 		}
 
 		var err error
 		xsite, err = ComputeXSite(c.infinispan, c.kube, siteService, c.log)
 		if err != nil {
-			c.log.Error(err, "Error in computeXSite configuration")
+			if errEv, ok := err.(*eventlog.ErrorEvent); ok {
```

Comment:
This doesn't seem like the right way to do things to me. If `xsite.go#appendKubernetesRemoteLocation` is supposed to return just an `error`, then we should just use `fmt.Errorf("%s: %w", reason, err)` in order to return a `error` directly from `appendKubernetesRemoteLocation`.

Reply:
this is to allow events when errors occurs [here](https://github.com/infinispan/infinispan-operator/pull/1008#issuecomment-845870043)
`ComputeXSite` doesn't accept a controller as argument so event can't be generated inside, also is used by the testsuite where there's no controller at all. So the idea was to minimize changes and generate events basing on the type of the returned error.

`error` is actually an interface and imo it's ok to return an object that implements it: I would say that `xsite.go#appendKubernetesRemoteLocation` returns "at least" an `error`

Reply:
> `error` is actually an interface and imo it's ok to return an object that implements it: I would say that `xsite.go#appendKubernetesRemoteLocation` returns "at least" an `error`

That's a fair point. I didn't realise that was what was happening.



> `ComputeXSite` doesn't accept a controller as argument so event can't be generated inside, also is used by the testsuite where there's no controller at all. So the idea was to minimize changes and generate events basing on the type of the returned error.

As we're already modifying the behaviour of `ComputeXsite` and `appendKubernetesRemoteLocation`, can't we just make it so that `ComputeXsite` takes the `EventLogger` interface and logs the event there? We can avoid using a specific `error` implementation and  just do the following in `config_controller.go`:

```go
if err != nil {
   reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
}
```

Reply:
@ryanemerson yep, it can be done, though the changes spread a little because there are some functions call between `ComputeXsite` and `appendKube...`. Also we need a `fake` EventLogger for the testsuite.
I don't have preference anyway I can change the implementation like that

Reply:
changed implementation